src/vscode-bicep/src/deploy/tree/registerTrees.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import { AzExtTreeDataProvider } from "@microsoft/vscode-azext-utils";
+
+import { ext } from "../../extensionVariables";
+import { AzLocationTreeItem } from "./AzLocationTreeItem";
+import { AzManagementGroupTreeItem } from "./AzManagementGroupTreeItem";
+import { AzResourceGroupTreeItem } from "./AzResourceGroupTreeItem";
+
+export function registerTrees(): void {
```

Comment:
Can we make this a `TreeManager` class instead? The trees can be initialized in a `register` method and can be exposed as properties. We can create a `TreeManager` instance and pass it to `DeployCommand` when initializing the extension. This should eliminate the need for `ext`.

Reply:
Another suggestion is to put the `tree` folder under `src` to maintain a flat project structure. In the future we may need to add tree items not related to the deploy command.

Reply:
> Can we make this a `TreeManager` class instead? The trees can be initialized in a `register` method and can be exposed as properties. We can create a `TreeManager` instance and pass it to `DeployCommand` when initializing the extension. This should eliminate the need for `ext`.

Thanks for the suggestion @shenglol. Refactored as part of this commit:
https://github.com/Azure/bicep/commit/62ec0b02cb7e0970c823c0b3f142ebe7fbc9b005

Reply:
> Another suggestion is to put the `tree` folder under `src` to maintain a flat project structure. In the future we may need to add tree items not related to the deploy command.

Updated here:
https://github.com/Azure/bicep/commit/4340d2abfe3b26cb27f47085b7ad9aa1e870f6a2

Thanks!